src/package/components/LessonOverviews/index.js

```diff
@@ -22,8 +22,8 @@ class LessonOverviews extends Component {
     currentPage: 1,
   }
 
-  handleCurrentPage = (currentPage) => {
-    this.setState({currentPage})
+  handleCurrentPage = (currentPage, request) => {
+    this.setState({currentPage}, () => {request()})
```

Comment:
Ensure request is only called after set state is complete, otherwise there is a race condition